packages/desktop-client/src/components/common/Button.tsx

```diff
@@ -1,33 +1,29 @@
-import React from 'react';
+import React, { forwardRef, type ReactNode } from 'react';
 
-import { css } from 'glamor';
-import type { CSSProperties } from 'glamor';
+import { css, type CSSProperties } from 'glamor';
 
 import Loading from '../../icons/AnimatedLoading';
 import { styles, colors } from '../../style';
+import { type HTMLPropsWithStyle } from '../../types/utils';
 
 import View from './View';
 
-interface ButtonProps
-  extends Omit<React.HTMLProps<HTMLButtonElement>, 'style'> {
+type ButtonProps = HTMLPropsWithStyle<HTMLButtonElement> & {
   pressed?: boolean;
   primary?: boolean;
   hover?: boolean;
   bare?: boolean;
   disabled?: boolean;
-  style?: CSSProperties;
   hoveredStyle?: CSSProperties;
   activeStyle?: CSSProperties;
+  textStyle?: CSSProperties;
   bounce?: boolean;
   as?: 'button';
-  children: React.ReactNode;
-  onClick?: () => void;
-}
+  children: ReactNode;
+  onClick?: (ev) => void;
```

Comment:
Should this be a React MouseEvent or similar?

Reply:
Actually not needed at all as already defined in HTMLButtonElement props